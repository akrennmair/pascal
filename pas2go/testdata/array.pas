program test;

var arr : array[1..5] of integer;
	i, j : integer;
	matrix : array[-5..+5, -3..+3] of real;

begin
	for i := 1 to 5 do
		arr[i] := i*i;

	for i := -5 to 5 do
		for j := 3 downto -3 do
		begin
			matrix[i, j] := 23.5
		end
end.
