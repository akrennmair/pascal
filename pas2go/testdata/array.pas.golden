package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		arr    [5]int
		i      int
		j      int
		matrix [11][7]float64
	)
	_ = arr
	_ = i
	_ = j
	_ = matrix

	for i = 1; i <= 5; i++ {
		arr[i-(1)] = i * i
	}
	for i = (-5); i <= 5; i++ {
		for j = 3; j >= (-3); j-- {
			matrix[i-(-5)][j-(-3)] = 23.5e0
		}
	}
}
